### DISCUSSION TITLE
Item Extensions

### DISCUSSION THREAD

**Discussion by claytongulick** [directus/directus#14093] [Feature Requests]
This is for the ability to add an extension to control the display of an item.

It would work in a similar way to how layouts work for collections, but would permit custom displays of items and sidebar.

Bonus functionality would be to add conditions that could drive the display of different item extensions, for example, in a patient table, if a patient status is "acute", a different item display extension would show than if the patient status was "discharged", or if a user's role is "Care Team Member" a different extension would be displayed than "Billing Manager".

**Comment by benhaynes** [comment]
I love this. While I'd love to make our native item forms as full-featured, customizable, and robust as possible... having this extensibility as an _escape hatch_ is a great idea. I can think of many opportunities to showcase item details in completely different (or proprietary) ways.

I agree that conditional logic to select a display would make this much more powerful, too.

We're starting in on a big push for extensions in about a month, which will include improving the dev experience, adding the extension build tools, improving the docs, adding new extension types, and (of course) building out the Marketplace. Would love to keep this on the radar for that.

**Reply by claytongulick** [reply]
We're happy to add this to the list of issues we're sponsoring, if that helps? Maybe check with Rob on the quote he's working up for us and see if it's feasible to add this?

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
